src/prefect/workers/base.py

```diff
@@ -511,6 +519,87 @@ def get_flow_run_logger(self, flow_run: "FlowRun") -> PrefectLogAdapter:
+
+        If `with_healthcheck` is set, the worker will start a healthcheck server which
+        can be used to determine if the worker is still polling for flow runs and restart
+        the worker if necessary.
+
+        Args:
+            run_once: If set, the worker will only run each loop once then return.
+            with_healthcheck: If set, the worker will start a healthcheck server.
+            printer: A `print`-like function where logs will be reported.
+        """
+        async with self as worker:
+            # wait for an initial heartbeat to configure the worker
+            await worker.sync_with_backend()
+            # schedule the scheduled flow run polling loop
+            async with anyio.create_task_group() as loops_task_group:
+                loops_task_group.start_soon(
+                    partial(
+                        critical_service_loop,
+                        workload=self.get_and_submit_flow_runs,
+                        interval=PREFECT_WORKER_QUERY_SECONDS.value(),
+                        run_once=run_once,
+                        jitter_range=0.3,
+                        backoff=4,  # Up to ~1 minute interval during backoff
+                    )
+                )
+                # schedule the sync loop
+                loops_task_group.start_soon(
+                    partial(
+                        critical_service_loop,
+                        workload=self.sync_with_backend,
+                        interval=self.heartbeat_interval_seconds,
+                        run_once=run_once,
+                        jitter_range=0.3,
+                        backoff=4,
+                    )
+                )
+                loops_task_group.start_soon(
+                    partial(
+                        critical_service_loop,
+                        workload=self.check_for_cancelled_flow_runs,
+                        interval=PREFECT_WORKER_QUERY_SECONDS.value() * 2,
+                        run_once=run_once,
+                        jitter_range=0.3,
+                        backoff=4,
+                    )
+                )
+
+                self._started_event = await self._emit_worker_started_event()
+
+                if with_healthcheck:
+                    from prefect.workers.server import start_healthcheck_server
+
+                    # we'll start the ASGI server in a separate thread so that
+                    # uvicorn does not block the main thread
+                    webserver_thread = threading.Thread(
+                        name="healthcheck-server-thread",
+                        target=partial(
+                            start_healthcheck_server,
+                            worker=self,
+                            query_interval_seconds=PREFECT_WORKER_QUERY_SECONDS.value(),
+                        ),
+                        daemon=True,
+                    )
+                    webserver_thread.start()
```

Comment:
I think we will want to kill this thread proactively in a `finally`, right?  If not, the thread won't exit until the process exits, which could lead to a memory leak if you were doing something interesting like starting & stopping multiple workers, etc.

Reply:
Good call! I added handling for that in 911b2dc9cbc638d0b7c6cc970309f2f4e93f7e6f

Reply:
Rad, looks great!